File: lab-resource-utilization/frontend/src/services/calibrationService.js
import api from "./api";

const calibrationService = {

    getAllCalibrations() {
        return api.get("/api/calibrations");
    },

    getCalibrationById(id) {
        return api.get(`/api/calibrations/${id}`);
    },

    addCalibration(data) {
        return api.post("/api/calibrations", data);
    },

    updateCalibration(id, data) {
        return api.put(`/api/calibrations/${id}`, data);
    },

    deleteCalibration(id) {
        return api.delete(`/api/calibrations/${id}`);
    },

    getCalibrationHistory(equipmentId) {
        return api.get(`/api/calibrations/equipment/${equipmentId}`);
    },

    getDueSoonCalibrations(days = 30) {
        return api.get(`/api/calibrations/due-soon?days=${days}`);
    },

    getExpiredCalibrations() {
        return api.get("/api/calibrations/expired");
    },

    // NEW METHODS
    getActiveCalibrations() {
        return api.get("/api/calibrations/active");
    },

    getFailedCalibrations() {
        return api.get("/api/calibrations/failed");
    },

    getSummary() {
        return api.get("/api/calibrations/summary");
    }

};

export default calibrationService;

File: lab-resource-utilization/src/main/java/com/example/lab_resource_utilization/service/CalibrationService.java
package com.example.lab_resource_utilization.service;

import com.example.lab_resource_utilization.dto.CalibrationRequest;
import com.example.lab_resource_utilization.dto.CalibrationResponse;

import java.util.List;
import java.util.Map;
public interface CalibrationService {

    // Add a new calibration record
    CalibrationResponse addCalibration(CalibrationRequest request);

    // Update an existing calibration record
    CalibrationResponse updateCalibration(Long id, CalibrationRequest request);

    // Delete a calibration record
    void deleteCalibration(Long id);

    // Get calibration by ID
    CalibrationResponse getCalibrationById(Long id);

    // Get all calibration records
    List<CalibrationResponse> getAllCalibrations();

    // Get calibration history of an equipment
    List<CalibrationResponse> getCalibrationHistory(Long equipmentId);

    // Get calibrations due within the next specified number of days
    List<CalibrationResponse> getDueSoonCalibrations(int days);

    // Get expired calibrations
    List<CalibrationResponse> getExpiredCalibrations();

    List<CalibrationResponse> getActiveCalibrations();

List<CalibrationResponse> getFailedCalibrations();

Map<String, Long> getCalibrationSummary();
}